### PR TITLE
use roles for compiler team maintainers

### PR DESCRIPTION
**Pull request by davidtwco**
- Allow alumni to have roles as the compiler team want to recognise the team member for the role they had while active
- Remove compiler team maintainer subteam and add these as roles to the team members

### CODE REVIEW

**Review by wesleywiser — APPROVED** [review]

**Review by rylev — DISMISSED** [review]

File: src/schema.rs [hunk 1]
```diff
@@ -478,6 +474,10 @@ impl Team {
         &self.people.members
     }
 
+    pub(crate) fn explicit_alumni(&self) -> &[TeamMember] {
```
[the remark this answers is not in the document]
Comment (davidtwco) [reply src/schema.rs:477]: It just mirrors the `members` naming - `explicit_members` returns the `TeamMember`s directly, and `members` returns an elaborated list of members as `&'a str`. It doesn't make sense to have a `alumni` equivalent of that, but I thought it made sense for `explicit_alumni` to return `TeamMember`s.

**Comment by Nadrieril** [comment]
Pinging this as it looks mergeable and would be a nice improvement.

**Comment by Mark-Simulacrum** [comment]
Happy to merge with a rebase.

**Comment by davidtwco** [comment]
Rebased!

**Review by Mark-Simulacrum — APPROVED** [review]

**Pull request merged**